from cli.rich_table import print_message

def input_error(func):
    def wrapper(*args, **kwargs):
        try:
            return func(*args, **kwargs)

        except ValueError as e:
            print_message(str(e) or "–ù–µ–ø—Ä–∞–≤–∏–ª—å–Ω–µ –∑–Ω–∞—á–µ–Ω–Ω—è.", style="bold red", title="‚ö†Ô∏è Force Disturbance")

        except KeyError as e:
            print_message(str(e) or "–ö–æ–Ω—Ç–∞–∫—Ç –Ω–µ –∑–Ω–∞–π–¥–µ–Ω–æ.", style="bold red", title="üõ∏ Missing Contact")

        except IndexError as e:
            print_message(str(e) or "–ù–µ–¥–æ—Å—Ç–∞—Ç–Ω—å–æ –∞—Ä–≥—É–º–µ–Ω—Ç—ñ–≤.", style="bold red", title="üì¶ Missing Data")

        except NameError as e:
            print_message(
                "Looks like something important wasn't defined... Is your code missing a variable or is your command incomplete?",
                style="bold red", title="üí• Critical Failure in the Code Matrix")

        except Exception as e:
            print_message(
                "Wokie encountered an unknown galactic anomaly.\n"
                f"[dim]({type(e).__name__}: {e})[/dim]",
                style="bold red", title="üí´ Unexpected Disturbance in the Force")

    return wrapper
